repository.go

```diff
@@ -77,35 +77,70 @@ type Repository struct {
 	wt billy.Filesystem
 }
 
-type InitOptions struct {
-	// The default branch (e.g. "refs/heads/master")
-	DefaultBranch plumbing.ReferenceName
+type initOptions struct {
+	defaultBranch plumbing.ReferenceName
+	workTree      billy.Filesystem
+	objectFormat  formatcfg.ObjectFormat
+}
+
+func newInitOptions() initOptions {
+	return initOptions{
+		defaultBranch: plumbing.Master,
+		workTree:      nil,
+		objectFormat:  "",
+	}
+}
+
+type InitOption func(*initOptions)
```

Comment:
For the purposes of this PR, we can leave this as-is. However, as we expand other parts of the code we may need to amend this so that we can enable the reuse of the same `With<Option>` across different use cases (e.g. `Init`, `Clone`, etc), without having to prefix them.

Let's leave this comment open for future reference.

Reply:
I had the same thought here, looking ahead to the possibility of future collisions, especially after initially running into them between `Init` and `PlainInit`.

One possibility that comes to mind is establishing a policy that all `With<Option>` functions need to be in a scoped package which helps guide clients into using the right option set for the package, e.g. `repo.WithWorkTree(...)` rather than `git.WithWorkTree(...)`

Reply:
The `--bare` option is one that may collide, for example, as both `git init --bare` and `git clone --bare` are supported by the CLI.